pkg/scheduler/preemption/preemption.go

```diff
@@ -333,10 +334,35 @@ func workloadFits(wlReq cache.FlavorResourceQuantities, cq *cache.ClusterQueue,
 				cohortResRequestable = cq.Cohort.RequestableResources[flvQuotas.Name]
 			}
 			for rName, rReq := range flvReq {
-				limit := flvQuotas.Resources[rName].Nominal
+				nominal := flvQuotas.Resources[rName].Nominal
+				// lendingLimit here represents how much quota can be lent from the cohort
+				var lendingLimit, borrowingLimit, limit int64
+				flagBorrow, flagLend := false, false
 				if flvQuotas.Resources[rName].BorrowingLimit != nil && allowBorrowing {
-					limit += *flvQuotas.Resources[rName].BorrowingLimit
+					flagBorrow = true
+					borrowingLimit = *flvQuotas.Resources[rName].BorrowingLimit
 				}
+				if features.Enabled(features.LendingLimit) && cq.Cohort != nil && allowBorrowing {
+					flagLend = true
+					if flvQuotas.Resources[rName].LendingLimit != nil {
+						// if LendingLimit is not nil here, we need to recalculate cohort.RequestableResources,
+						// because we are borrowing from the cohort now, need to count our own nominal quota in.
+						cohortResRequestable[rName] += flvQuotas.Resources[rName].Nominal - *flvQuotas.Resources[rName].LendingLimit
```

Comment:
This is not right, because you're changing the underlying value of the cache, which is used through the whole scheduling cycle.